frontend/projects/upgrade/src/app/features/dashboard/home/components/factorial-experiment-design/factorial-experiment-design.component.ts

```diff
@@ -473,22 +471,30 @@ export class FactorialExperimentDesignComponent implements OnInit, OnChanges, On
     }
   }
 
-  validateFactorNames(factors: ExperimentFactorFormData[]) {
+  isFactorTableRowValid(factorRowIndex: number): boolean {
+    const factors = this.factorialExperimentDesignForm.value.factors;
+    const factorRow = factors[factorRowIndex];
     this.factorPointErrors = [];
-    this.levelPointErrors = [];
-    // Used to differentiate errors
-    const duplicateFactors = [];
-
-    factors.forEach((factor) => {
-      this.validateLevelNames(factor.levels, factor.name);
-    });
+    if (!factorRow.name?.trim()) {
+      return false;
+    }
+    if (factors.some((factor, index) => factor.name === factorRow.name && index !== factorRowIndex)) {
+      this.factorPointErrors.push('Factor should be unique');
+    }
+    return !this.factorPointErrors.length;
+  }
 
-    // Factor Points error messages
-    if (duplicateFactors.length === 1) {
-      this.factorPointErrors.push(duplicateFactors[0] + this.factorErrorMessages[0]);
-    } else if (duplicateFactors.length > 1) {
-      this.factorPointErrors.push(duplicateFactors.join(', ') + this.factorErrorMessages[1]);
+  isLevelTableRowValid(levelRowIndex: number, factorIndex: number): boolean {
+    const levels = this.factorialExperimentDesignForm.value.factors[factorIndex].levels;
```

Comment:
this is fine of course, just wanted to point out a convenience function on this component, since this is :

`this.getLevels(factorIndex)` can be used instead of `this.factorialExperimentDesignForm.value.factors[factorIndex].levels` if you so please